src/Microsoft.ML.CpuMath/AvxIntrinsics.cs

```diff
@@ -951,8 +943,6 @@ public static unsafe void Scale(float scale, Span<float> dst)
 
         public static unsafe void ScaleSrcU(float scale, ReadOnlySpan<float> src, Span<float> dst, int count)
         {
-            Contracts.Assert(src.Length == dst.Length);
```

Comment:
I think asserts here are still valuable, aren't they?

Contracts.Assert(count <= src.Length  && src.Length <= dst.Length);

Reply:
I will add the more appropiate asserts with the documentation pr.
I  am thinking of an another approach of slicing the span to the required length before passing to these functions and adding the ```contracts.Assert(src.Length == dst.Length)``` . Is this approach worth doing ?

Reply:
IMO - no I don't think that approach is worth doing. That's why the `count` parameter is passed to these functions - so the spans don't need to be the same length, and one of the spans isn't chosen as "the one with the right length".  It's unnecessary logic to ensure the spans are the same length.

Slicing spans is cheap, but it isn't free. You still have bounds checks to ensure the sliced values are less than the span length.